app/uk/gov/hmrc/plasticpackagingtax/registration/controllers/ContactDetailsConfirmAddressController.scala

```diff
@@ -104,6 +84,16 @@ class ContactDetailsConfirmAddressController @Inject() (
       )
     )
 
+  private def getIncorporationDetails()(implicit
+    request: JourneyRequest[AnyContent]
+  ): Future[Address] =
+    request.registration.organisationDetails.incorporationDetails.fold(
+      throw DownstreamServiceError(
+        "Failed to fetch incorporated details from cache",
+        RegistrationException("Failed to fetch incorporated details from cache")
+      )
+    )(details => updateBusinessAddress(details.companyAddress.toPptAddress))
```

Comment:
I don't like methods which have side-effects (updating the address in the cache) which are not suggested by their names.